scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/GraphQLMerlinService.java

```diff
@@ -879,6 +929,75 @@ public Optional<SimulationResults> getSimulationResults(PlanMetadata planMetadat
     }
   }
 
+  public record DatasetMetadata(DatasetId datasetId, Duration offsetFromPlanStart){};
+
+  public Optional<List<DatasetMetadata>> getExternalDatasets(final PlanId planId)
+  throws PlanServiceException, IOException
+  {
+    final var datasets = new ArrayList<DatasetMetadata>();
+    final var request = """
+        query {
+          plan_dataset(where: {plan_id: {_eq: %d}, simulation_dataset_id: {_is_null: true}}) {
```

Comment:
Suggestion: Order the results by `dataset_id`. This will mean that if there are any overridden values between datasets, the final value in the ExternalProfiles object fetched in `getExternalProfiles` will always be the version from the most recent dataset.
```suggestion
          plan_dataset(where: {plan_id: {_eq: %d}, simulation_dataset_id: {_is_null: true}}, order_by: {dataset_id:asc}) {
```